src/filesystem_common.cpp

```diff
@@ -122,9 +122,14 @@ bool blacklist_pattern_list::match_dir(const std::string& name) const
 					   std::bind(&utils::wildcard_string_match, std::ref(name), std::placeholders::_1));
 }
 
-std::string get_prefs_file()
+std::string get_synced_prefs_file()
 {
-	return get_user_config_dir() + "/preferences";
+	return get_sync_dir() + "/preferences";
+}
+
+std::string get_unsynced_prefs_file()
+{
+	return get_user_data_dir() + "/preferences";
```

Comment:
```suggestion
	return get_config_data_dir() + "/preferences";
```
It makes no sense to just change this location. Either dissolve the user config dir or keep it as is.

Reply:
I've already said how I intend to handle this multiple times now. I don't have anything else to say about it.

Reply:
So this is intentionally doing part of the changes of a later PR? What's the point of that? I do not understand the insistence on doing changes in a confusing way. Even changes that affect people working on wesnoth by moving important files around without any mention in the commit message or PR description.

I'm _very_ opposed to this. I read commits all the time. I do want wesnoth's commit history to be as easy to read and follow as possible. I'm happy to do whatever changes myself if this is about being more work or something.

Reply:
There, done.